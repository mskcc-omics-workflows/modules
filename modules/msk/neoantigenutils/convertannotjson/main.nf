process NEOANTIGENUTILS_CONVERTANNOTJSON {
    tag "$meta.id"
    label 'process_single'
    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
        'docker://mskcc/neoantigen-utils-base:1.0.0':
        'docker.io/mskcc/neoantigen-utils-base:1.0.0' }"

    input:
    tuple val(meta),  path(annotatedJSON)

    output:
    tuple val(meta), path("*.tsv"),              emit: neoantigenTSV
    path "versions.yml",                         emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    def args = task.ext.args ?: ''
    def prefix = task.ext.prefix ?: "${meta.id}"
    """
        convertannotjson.py \
            --json_file ${annotatedJSON} \
            --output_file ${prefix}_neoantigens.tsv

        cat <<-END_VERSIONS > versions.yml
        "${task.process}":
            convertannotjson: \$(echo \$(convertannotjson.py -v))
        END_VERSIONS
    """

    stub:
    def args = task.ext.args ?: ''
    def prefix = task.ext.prefix ?: "${meta.id}"
    """

        touch ${prefix}_neoantigens.tsv

        cat <<-END_VERSIONS > versions.yml
        "${task.process}":
            convertannotjson: \$(echo \$(convertannotjson.py -v))
        END_VERSIONS
    """
}
