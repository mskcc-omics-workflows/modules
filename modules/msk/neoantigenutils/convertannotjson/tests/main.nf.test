nextflow_process {

    name "Test Process NEOANTIGENUTILS_CONVERTANNOTJSON"
    script "../main.nf"
    process "NEOANTIGENUTILS_CONVERTANNOTJSON"

    tag "modules"
    tag "modules_nfcore"
    tag "neoantigenutils"
    tag "neoantigenutils/convertannotjson"
    tag "modules_msk"

    test("neoantigenutils_convertannotjson - output(test) - tsv") {

        when {

            process {
                """

                input[0] = [
                    [ id:'test'], // meta map
                    file(params.test_data_mskcc['neoantigen']['test_annotated'], checkIfExists: true)
                    ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match()}
            )
        }

    }

    test("neoantigenutils_convertannotjson - output(test) - tsv - stub") {

        options "-stub"

        when {

            process {
                """

                input[0] = [
                    [ id:'test'], // meta map
                    file('test_annotated')
                    ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }
}
