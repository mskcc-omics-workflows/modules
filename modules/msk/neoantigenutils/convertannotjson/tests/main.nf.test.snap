{
    "neoantigenutils_convertannotjson - output(test) - tsv": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "test_neoantigens.tsv:md5,4931fb72bba8bbeb3bc6cef19b99b01a"
                    ]
                ],
                "1": [
                    "versions.yml:md5,e36ea44c5cc6130f40b1a100f7e84fb1"
                ],
                "neoantigenTSV": [
                    [
                        {
                            "id": "test"
                        },
                        "test_neoantigens.tsv:md5,4931fb72bba8bbeb3bc6cef19b99b01a"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,e36ea44c5cc6130f40b1a100f7e84fb1"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.04.2"
        },
        "timestamp": "2024-07-24T16:49:37.133693"
    },
    "neoantigenutils_convertannotjson - output(test) - tsv - stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "test_neoantigens.tsv:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "1": [
                    "versions.yml:md5,e36ea44c5cc6130f40b1a100f7e84fb1"
                ],
                "neoantigenTSV": [
                    [
                        {
                            "id": "test"
                        },
                        "test_neoantigens.tsv:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,e36ea44c5cc6130f40b1a100f7e84fb1"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.04.2"
        },
        "timestamp": "2024-07-24T16:49:41.629164"
    }
}